### DISCUSSION TITLE
Controlling formatting in a Table

### DISCUSSION THREAD

**Discussion by runrc** [cogentcore/core#1333] [Q&A]
We would like the ability to be able to control the formatting of values in a Table. 

For example,
1)  consider a column of type `time.Time`, how can we control the formatting of column values?
2) consider a custom user type, how is the formatting / displaying of such values controlled?
3) consider a floating type, we'd like to format values with certain significant figures / decimal places, how can this be done?
4) consider a currency, we'd like to show values with currency pre(post) fixes, how can this be done?

**Comment by kkoreilly** [comment]
1. We will add support for that via struct tags soon.
2. You can make a custom widget type that implements the `WidgetValue` function and uses `Init` and/or `OnBind` to specify the display. You can then call `core.AddValueType` to register your custom widget with your custom user type. We will have more organized documentation on this soon, but you can see the code for [FileButton](https://github.com/cogentcore/core/blob/content/core/filepicker.go#L712) and [TimeInput](https://github.com/cogentcore/core/blob/content/core/timepicker.go#L308) as good examples of this approach.
3. Please try the `format` struct tag, which specifies the formatting of numbers in `fmt` style (ex: `format:"%.3g"`).
4. If there is a unique type for the currency, I would try the same custom widget type approach as for your second example.

Another good approach for most of those use cases will be `CoreFieldValue` once I implement it, likely next week (#1204).
